quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/legacy/infinispan/CacheManagerFactory.java

```diff
@@ -86,6 +97,10 @@ private DefaultCacheManager startCacheManager() {
         return new DefaultCacheManager(builder, isStartEagerly());
     }
 
+    private boolean isMetricsEnabled() {
```

Comment:
We can avoid looking up if a build option is set by resolving it at build time. For that, you can change this line to create the `CacheManagerFactory` instance from the recorded by passing the value for the `metrics-enabled` option.

Sounds like a nitpick but we need to try to reduce any unnecessary steps during bootstrapping and favor building the configuration during re-augmentation / build time.

Looking up options at runtime is a bit expensive as it ends up going through all MP config sources, etc.

Reply:
Thanks for your suggestions! I applied them (and learned something).